src/python/zquantum/vqe/singlet_uccsd_test.py

```diff
@@ -0,0 +1,124 @@
+from zquantum.core.interfaces.ansatz_test import AnsatzTests
+from zquantum.core.circuit import Circuit
+from .singlet_uccsd import SingletUCCSDAnsatz
+import unittest
+import numpy as np
+
+
+class TestSingletUCCSDAnsatz(unittest.TestCase, AnsatzTests):
+    def setUp(self):
+        self.number_of_layers = 1
+        self.number_of_alpha_electrons = 1
+        self.number_of_spatial_orbitals = 3
+        self.transformation = "Jordan-Wigner"
+
+        self.ansatz = SingletUCCSDAnsatz(
+            number_of_layers=self.number_of_layers,
+            number_of_spatial_orbitals=self.number_of_spatial_orbitals,
+            number_of_alpha_electrons=self.number_of_alpha_electrons,
+            transformation=self.transformation,
+        )
+
+    def test_init_asserts_number_of_spatial_orbitals(self):
+        # Given
+        incorrect_number_of_spatial_orbitals = 0
+
+        # When/Then
+        with self.assertRaises(ValueError):
+            self.ansatz = SingletUCCSDAnsatz(
+                number_of_layers=self.number_of_layers,
+                number_of_spatial_orbitals=incorrect_number_of_spatial_orbitals,
+                number_of_alpha_electrons=self.number_of_alpha_electrons,
+                transformation=self.transformation,
+            )
+
+    def test_set_number_of_alpha_electrons(self):
+        # Given
+        new_number_of_alpha_electrons = 2
+
+        # When
+        self.ansatz.number_of_alpha_electrons = new_number_of_alpha_electrons
+
+        # Then
+        self.assertEqual(
+            self.ansatz._number_of_alpha_electrons, new_number_of_alpha_electrons
+        )
+
```

Comment:
I think the correct assertion is `self.assertEqual(self.ansatz.number_of_alpha_electrons, new_number_of_alpha_electrons)`. That's because the test probably should assert that "if I set number of alpha electrons, I can read back the same value". The fact that it is stored in `_number_of_alpha_electrons` is irrelevant from the perspective of clients of this class.

Reply:
Fixed in [58a82b](https://github.com/zapatacomputing/z-quantum-vqe/pull/5/commits/58a82b05a9c26f2fb4654511ae21891ef1d1ee15).